SwissDataScienceCenter/renku-ui
# feat: add ui server authentication

This big PR is now ready to be reviewed!

In short, it adds all the logic for authenticating through the ui server. That required implementing the ui server infrastructure, adding a redis store, fixing the required values and routes.
The login and logout use the ui-server routes `/ui-server/auth/<endpoint>`. Those redirect to the old endpoints to trigger the backend services authentication (from the next release it's only GitLab).
Mind that the APIs aren't going through the ui-server yet. Let's see if the login/logout logic works, then we can easily route all the gateway requests through the ui-server.
For testing purposes, one can already test the `ui-server/api/*` routes. Currently, it re-routes all the requests targeting `ui-server/api/projects*` to the gateway, while all the others just return a recap of the sent headers (ofc the tokens are removed).
Another session cookie `ui-server-session` has been created. The old `session` cookie will be removed at a later point.

***How to test***
Since the automatic PR deployment won't have all the required values, I deployed a test version at
https://lorenzotest.dev.renku.ch

- Login/logout: Just use the normal buttons in the UI. Mind in the network tab that the requests go to `/ui-server/auth/*`.
- You can check if the keycloak authentication token would be added to the API by invoking https://lorenzotest.dev.renku.ch/ui-server/api/test . The `Authorization` header should be there for authenticated requests. Mind that the `ui-server-session` cookie is never removed once created.
- You can check if the authentication is working properly by invoking any GitLab API on the route `ui-server/api/projects*`. An example would be https://lorenzotest.dev.renku.ch/ui-server/api/projects?starred=true
Mind that the response will contain starred project only for authenticated users. Otherwise, it will show a list of public projects.

***Before merging***
Some changes may be required to the acceptance tests, and to other components as well. We also need to tag a gateway release (the current `master` is required).
Once the PR is ready, we should address those requirements and try to run the acceptance test with the `deploy` command.

/deploy renku=ui1337-uiserver-values
re #1337
